tests/components/snmp/test_integer_switch.py

```diff
@@ -0,0 +1,62 @@
+"""SNMP switch tests."""
+
+from unittest.mock import patch
+
+from pysnmp.hlapi import Integer32
+
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+config = {
+    SWITCH_DOMAIN: {
+        "platform": "snmp",
+        "host": "192.168.1.32",
+        # ippower-mib::ippoweroutlet1.0
+        "baseoid": "1.3.6.1.4.1.38107.1.3.1.0",
+        "payload_on": 1,
+        "payload_off": 0,
+    },
+}
+
+
+async def test_snmp_integer_switch_off(hass: HomeAssistant) -> None:
+    """Test snmp switch returning int 0 for off."""
+
+    mock_data = Integer32(0)
+    with patch(
+        "homeassistant.components.snmp.switch.getCmd",
+        return_value=(None, None, None, [[mock_data]]),
+    ):
+        assert await async_setup_component(hass, SWITCH_DOMAIN, config)
+        await hass.async_block_till_done()
+        state = hass.states.get("switch.snmp")
+        assert state.state == "off"
+
+
+async def test_snmp_integer_switch_on(hass: HomeAssistant) -> None:
+    """Test snmp switch returning int 1 for on."""
+
+    mock_data = Integer32(1)
+    with patch(
+        "homeassistant.components.snmp.switch.getCmd",
+        return_value=(None, None, None, [[mock_data]]),
+    ):
+        assert await async_setup_component(hass, SWITCH_DOMAIN, config)
+        await hass.async_block_till_done()
+        state = hass.states.get("switch.snmp")
+        assert state.state == "on"
+
+
+async def test_snmp_integer_switch_unknown(hass: HomeAssistant) -> None:
```

Comment:
Can we use caplog to check the warning message?

Reply:
done in 6dda938c53 - but I'm not entirely happy with it:
`homeassistant.components.snmp.switch:switch.py:280 Invalid payload '3' received for entity None, state is unknown\n"`

I don't know enough about entity lifecycle to know if the entity id is available that early on from somewhere else? subsequent warnings are fine for actual devices:
```
2024-08-06 20:15:10.625 WARNING (MainThread) [homeassistant.components.snmp.switch] Invalid payload '1' received for entity None, state is unknown
2024-08-06 20:15:40.632 WARNING (MainThread) [homeassistant.components.snmp.switch] Invalid payload '1' received for entity switch.9858dx_outlet_4, state is unknown
2024-08-06 20:16:10.632 WARNING (MainThread) [homeassistant.components.snmp.switch] Invalid payload '1' received for entity switch.9858dx_outlet_4, state is unknown
```

Reply:
Depends if they are loaded into hass

Reply:
my reading of https://developers.home-assistant.io/docs/core/entity/ and https://developers.home-assistant.io/docs/device_registry_index/ is that it hasn't yet been populated into the device registry, so there's no entity_id to get.

Reply:
entity_id is not directly related with device registry. as in, yes, the device registry affects the entity_id (when `_attr_has_entity_name = True`, but it's not required for it to receive one